### PR TITLE
Add EditCommand Feature

### PR DESCRIPTION
**Pull request by aaronxujiachen**
It allows users to edit their previous entries.

### CODE REVIEW

File: src/main/java/seedu/nuscents/commands/EditCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+    
+    @Override
+    public void execute(TransactionList transactionList) throws NuscentsException {
+        if (expense == null && allowance == null) {
+            throw new NuscentsException("Please enter either an expense or allowance to edit");
+        } else if (expense != null) {
+            transactionList.editTransation(this.index,this.expense);
+        } else{
+            transactionList.editTransation(this.index,this.allowance);
```
Comment (lckjosh) [inline src/main/java/seedu/nuscents/commands/EditCommand.java:30]: Is this check necessary? From what I understand this check should already be done in Parser.java

File: src/main/java/seedu/nuscents/parser/Parser.java [hunk 2]
```diff
@@ -78,6 +80,8 @@ public static <TransactionList> Command parseCommand(String text, TransactionLis
                 return new ViewCommand(parseTaskIndex(arguments));
             case COMMAND_FILTER:
                 return new FilterCommand(parseFilterCategory(arguments));
+            case COMMAND_EDIT:
+                return parseEdit(arguments);
```
Comment (lckjosh) [inline src/main/java/seedu/nuscents/parser/Parser.java:84]: Perhaps you could follow the format of the other commands, that is, return new EditCommand(parseEdit(arguments))

File: src/main/java/seedu/nuscents/parser/Parser.java [hunk 3]
```diff
@@ -304,6 +308,26 @@ public static String parseFind(String arguments) throws NuscentsException {
...
+            }
+            int index = parseTaskIndex(editDetails[0]);
+            String type = editDetails[1];
+            if (type.equals("expense")){
+                return new EditCommand(parseExpense(editDetails[2]), index);
+            } else if (type.equals("allowance")) {
+                return new EditCommand(parseAllowance(editDetails[2]), index);
+            } else{
+                throw new NuscentsException(MESSAGE_FATAL_ERROR);
+            }
```
Comment (spaceman03) [inline src/main/java/seedu/nuscents/parser/Parser.java:327]: I think there is a missing space after the "else"

File: src/main/java/seedu/nuscents/commands/EditCommand.java (hunk 1 above)
Comment (spaceman03) [inline src/main/java/seedu/nuscents/commands/EditCommand.java:30]: Missing space after "else"

**Pull request merged**